packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -692,16 +688,16 @@ class DevFS {
 /// Requires that the file system is the same for both the tool and application.
 class LocalDevFSWriter implements DevFSWriter {
   LocalDevFSWriter({
-    @required FileSystem fileSystem,
+    required FileSystem fileSystem,
   }) : _fileSystem = fileSystem;
 
   final FileSystem _fileSystem;
 
   @override
-  Future<void> write(Map<Uri, DevFSContent> entries, Uri baseUri, [DevFSWriter parent]) async {
+  Future<void> write(Map<Uri, DevFSContent> entries, Uri baseUri, [DevFSWriter? parent]) async {
     try {
       for (final Uri uri in entries.keys) {
-        final DevFSContent devFSContent = entries[uri];
+        final DevFSContent devFSContent = entries[uri]!;
```

Comment:
Can the `for` be rewritten as a `forEach` without the `!`?

Reply:
This block is asynchronous, with the `await devFSContent.contentsAsBytes()`. I think if we move it into a forEach, we might start writing to destination from one iteration before the previous Future has completed?

Reply:
You mean it wouldn't work with an async `forEach` block? And the `continue` becoming `return`?
```dart
entries.forEach((Uri uri, DevFSContent devFSContent) async { 
```